sdk/core/azure-core/src/main/java/com/azure/core/util/paging/ContinuablePagedByIteratorBase.java

```diff
@@ -49,10 +61,14 @@ public E next() {
     @Override
     public boolean hasNext() {
         // Request next pages in a loop in case we are returned empty pages for the by item implementation.
-        while (!done && needToRequestPage()) {
+        while (!done && needToRequestPage() && pageRetriever != null) {
             requestPage();
         }
 
+        while (!done && needToRequestPage() && syncPageRetriever != null) {
```

Comment:
The pageRetriever (sync or async) will be final once this instance is created. So, we don't need to check if it's null in a `while` loop.